src/main/java/jenkins/plugins/git/traits/BranchDiscoveryTrait.java

```diff
@@ -146,6 +148,14 @@ public String getDisplayName() {
             public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> originClass) {
                 return SCMHeadOrigin.Default.class.isAssignableFrom(originClass);
             }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public boolean isApplicableToHead(@NonNull Class<? extends SCMHeadMixin> headClass) {
+                return super.isApplicableToHead(headClass) && !(TagSCMHead.class.isAssignableFrom(headClass));
```

Comment:
for historical reasons, because Git plugin does not implement its own SCMHead for branches, we have to use this more complex check to exclude this authority from being applied to tags